src/components/expanded-state/unique-token/NFTBriefTokenInfoRow.tsx

```diff
@@ -22,48 +27,64 @@ const getIsFloorPriceSupported = (network: Network) => {
   }
 };
 
+const formatPrice = (
+  price: number | null | undefined,
+  tokenSymbol: string | null | undefined = 'ETH'
+) => {
+  if (price === null || price === undefined || !tokenSymbol) return NONE;
+  return `${price === 0 ? '< 0.001' : price} ${tokenSymbol}`;
+};
+
 export default function NFTBriefTokenInfoRow({
-  currentPrice,
-  lastPrice,
-  lastSalePaymentToken,
-  network: assetNetwork,
-  urlSuffixForAsset,
+  asset,
 }: {
-  currentPrice?: number | null;
-  lastPrice?: number | null;
-  lastSalePaymentToken?: string | null;
-  network: Network;
-  urlSuffixForAsset: string;
+  asset: UniqueAsset;
 }) {
   const { colors } = useTheme();
 
   const { navigate } = useNavigation();
 
-  const { nativeCurrency, network } = useAccountSettings();
+  const { nativeCurrency } = useAccountSettings();
+
+  const [floorPrice, setFloorPrice] = useState<string | null>(
+    asset?.floorPriceEth ? formatPrice(asset?.floorPriceEth, 'ETH') : null
+  );
 
-  const [floorPrice, setFloorPrice] = useState<string | null>(null);
+  const { data: listing } = useNFTListing({
+    contractAddress: asset?.asset_contract?.address ?? '',
+    tokenId: asset?.id,
+    network: asset?.network,
+  });
```

Comment:
Do we need to handle the error that could be returned here?

Reply:
I don't think so, except maybe a log if you think that'd be useful. If we can't find a listing we're just going to show the last sale price to the user anyway, so I don't think there's a need to communicate this error to them.